tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh

```diff
@@ -143,6 +143,64 @@ install_cached_component() {
 	mv "${component_tarball_name}" "${component_tarball_path}"
 }
 
+# We've to add a different cached function here as for using the shim-v2 caching
+# we have to rely and check some artefacts coming from the cc-rootfs-image and the
+# cc-tdx-rootfs-image jobs.
+install_cached_cc_shim_v2() {
+	local component="${1}"
+	local jenkins_build_url="${2}"
+	local current_version="${3}"
+	local current_image_version="${4}"
+	local component_tarball_name="${5}"
+	local component_tarball_path="${6}"
+	local root_hash_vanilla="${repo_root_dir}/tools/osbuilder/root_hash_vanilla.txt"
+	local root_hash_tdx="${repo_root_dir}/tools/osbuilder/root_hash_tdx.txt"
+
+	local rootfs_image_cached_root_hash="${jenkins_url}/job/kata-containers-2.0-rootfs-image-cc-$(uname -m)/${cached_artifacts_path}/root_hash_vanilla.txt"
+	local tdx_rootfs_image_cached_root_hash="${jenkins_url}/job/kata-containers-2.0-rootfs-image-tdx-cc-$(uname -m)/${cached_artifacts_path}/root_hash_tdx.txt"
+
+
+	wget "${rootfs_image_cached_root_hash}" -O "rootfs_root_hash_vanilla.txt" || return 1
+	if [ -f "${root_hash_vanilla}" ]; then
+		# There's already a pre-existent root_hash_vanilla.txt,
+		# let's check whether this is the same one cached on the
+		# rootfs job.
+
+		# In case it's not the same, let's proceed building the
+		# shim-v2 with what we have locally.
+		diff "${root_hash_vanilla}" "rootfs_root_hash_vanilla.txt" > /dev/null || return 1
+	fi
+	mv "rootfs_root_hash_vanilla.txt" "${root_hash_vanilla}"
+
+	wget "${rootfs_image_cached_root_hash}" -O "rootfs_root_hash_tdx.txt" || return 1
+	if [ -f "${root_hash_tdx}" ]; then
+		# There's already a pre-existent root_hash_tdx.txt,
+		# let's check whether this is the same one cached on the
+		# rootfs job.
+
+		# In case it's not the same, let's proceed building the
+		# shim-v2 with what we have locally.
+		diff "${root_hash_tdx}" "rootfs_root_hash_tdx.txt" > /dev/null || return 1
+	fi
+	mv "rootfs_root_hash_tdx.txt" "${root_hash_tdx}"
+
+	wget "${jenkins_build_url}/root_hash_vanilla.txt" -O "shim_v2_root_hash_vanilla.txt" || return 1
```

Comment:
Again - just to check I understand:
The above 'installs' to `${root_hash_vanilla}`/`${root_hash_tdx}` gets the version of the root hash cached with the rootfs and  `shim_v2_root_hash_vanilla` gets the versions that were cached with the shim-v2 and if either of them are out of date/don't match we re-build the shimv2 so it picks up the new hash into the kernel_params config?

Reply:
Okay, let me write down the whole workflow here as it will help us to understand better if I'm doing something wrong.

* We download the `root_hash_{vanilla,tdx}.txt from the **rootfs image build**.
* If we do have a `tools/osbuilder/root_hash_{tdx,vanilla}.txt`
  * We compare the downloaded one with the one from the system
    * If that's equal, we proceed
    * Otherwise, we force the shim-v2 build.
* Otherwise we just move the download files to `tools/osbuilder/root_hash_{tdx,vanilla}.txt`

* We download the `root_hash_{vanilla,tdx}.txt` from the **shim-v2 build**
* If the ones from the rootfs build and from the shim-v2 build are  **not** the same, we for the shim-v2 build
* Otherwise, we go ahead with the `intsall_cache_component` which will then check things like
  * last commit on the `src/runtime` directory
  * go version used to perform that build
  * rust version used to perform that build
  * cached build docker image to perform that build
* If any of those things are not the same, we force a shim-v2 build
* Otherwise we use a cached version of shim-v2

Doies the workflow make sense, @stevenhorsman?

Reply:
Yep - thanks for spelling out the flow in more detail and with more clarity than my long sentence above!